src/components/Cart/CartListItem.jsx

```diff
@@ -0,0 +1,141 @@
+import { useState } from 'react';
+import { useAuth, useCart } from "contexts/";
+import { updateProductInCart, deleteProductInCart } from 'utils/'
+import { useToast } from "custom-hooks/useToast";
+
+const CartListItem = ({ cartItem }) => {
+    const [isOngoingNetworkCall, setIsOngoingNetworkCall] = useState(false); 
+    const {
+        author,
+        bookType,
+        coverImg,
+        discountPercent,
+        _id,
+        sellingPrice,
+        title,
+        qty
+    } = cartItem;
+ 
+    const localeSellingPrice = sellingPrice.toLocaleString("en-IN", {minimumFractionDigits: 2, maximumFractionDigits: 2});
+
+    const { authState: { token } } = useAuth();
+    const { showToast } = useToast();
+    const { cartDispatch } = useCart();
+
+    const handleCartQuantityButtonClick = async event => {
+        const { value } = event.target;
+        let operation = '';
+        switch(value) {
+            case 'DECREASE_QUANTITY':
+                operation = 'decrement';
+                break;
+            case 'INCREASE_QUANTITY':
+                operation = 'increment'
+                break;
+            default: 
+                operation = '';
+                break;
+        }
+
+        if(qty >= 3 && operation === 'increment') {
+            showToast('We are sorry! Only 3 units allowed in each order!', 'error');
+            return;
+        }
+
+        try {
+            setIsOngoingNetworkCall(true);
+            const productUpdatedInCart = await updateProductInCart(_id, token, operation);
+
+            if(productUpdatedInCart) {
+                showToast('Quantity updated successfully!', 'success');
+                cartDispatch({type: 'ADD_TO_CART', payload: {cartItems: productUpdatedInCart, error: false, loading: false}});
+            }
+            else {
+                showToast('Failed to update quantity. Please try again later.', 'error');
+            }
+            setIsOngoingNetworkCall(false);
+        }
+        catch(error) {
+            console.log('Something went really wrong!');
+            setIsOngoingNetworkCall(false);
+        }        
+    }
+
+    const handleRemoveFromCart = async () => {
+        try {
+            const productDeletedInCart = await deleteProductInCart(_id, token);
+            if(productDeletedInCart) {
+                showToast('Product deleted successfully!', 'success');
```

Comment:
Can we use these messages in constants and use here

Reply:
We could if we had the same messages used for all the cases. But since the messages are different in every case, I thought it did not make sense to create constants. Maybe for the themes, it makes sense.